zsh/functions/clear_port

```diff
@@ -0,0 +1,11 @@
+# Kills the process running on the provided port
+#
+# clear_port 3000
+
+clear_port() {
+  if [ "$1" ]; then
+    kill "$(lsof -t -i4TCP:"$1")"
+    return
+  fi
+  echo >&2 "Port number needed to determine process id"
```

Comment:
We should probably pass failure back to the shell (`exit 1`) so it knows it's an error, both here and in `whats_in_port`.

We can get fancier, too: don't kill if we can't find it, like so:

```sh
if [ -n "$1" ]; then
  port_num="$(lsof -ti4TCP:"$1")"
  if [ $? -eq 0 ]; then
    kill "$port_num"
  fi
else
  echo >&2 Usage: clear_port port-number
  exit 1
fi
```

Reply:
So I tried to add this, but it ends up closing my current terminal tab when I add the `exit 1`. Is there a way to do this that doesn't do that?

Reply:
Yes:

> Why is this a function? What do you think of making it a program in bin, available to all our shell users and not just Zsh users?

Reply:
Though if you want to keep this a function you could probably use `return` instead of `exit`. Though I don't know why this should be a function.

Reply:
Ah I missed that. What does "program in bin" mean?

Reply:
Move these two files to `bin/`, add `#!/bin/sh` to the top, and remove the function declaration around the body (so remove `clear_port() {` and `whats_in_port() {`).

Shell functions aren't very useful but if you need e.g. a program that will exit your shell or change directories on your behalf then they're quite handy. They're incredibly powerful, though, at the expense of being non-portable and bug-prone.

Reply:
Ah I see and then they'll be in the path and can be invoked similarly. I was missing the forest through the trees, thanks!

Reply:
Done 12cf89a